### ISSUE TITLE
Invalid Serverless Application Specification document error while using OwnershipControls property with the S3 bucket specified as BucketTrigger for AWS::Serverless::Function resource

### ISSUE THREAD

**Issue by akshatki** [aws/serverless-application-model#1879]
**Description:**
When using 'OwnershipControls' property for 'AWS::S3::Bucket' type resource and then using this bucket resource as BucketTrigger within the Events property of an 'AWS::Serverless::Function', stack creation fails with the error:

"Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [<S3-bucket-logical-ID>] is invalid. property OwnershipControls not defined for resource of type AWS::S3::Bucket. Rollback requested by user."

**Steps to reproduce the issue:**
Template to reproduce this issue is included at : https://gist.github.com/akshatki/61765ca285ce6c7893f2ebda070e6862

Launch this template from the Cloudformation console with default settings.

**Observed result:**
This 'Invalid Serverless Application Specification document' error message is received post which stack rollback begins:

"Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [<S3-bucket-logical-ID>] is invalid. property OwnershipControls not defined for resource of type AWS::S3::Bucket. Rollback requested by user."

Also this error is not faced if the OwnershipControls property block is removed from the S3 bucket resource.

**Expected result:**

Expected the template to be validated, bucket to be created with OwnershipControls property as specified and then expected this bucket to be used as BucketTrigger value for the serverless resource.

Additionally, for the time being it would be really helpful if a suitable workaround is provided.

**Comment by hawflau** [comment]
@akshatki Thanks for raising the issue and providing the template to reproduce the issue!
It seems `OwenershipControls` is a new property that we haven't updated in `samtranslator/model/s3.py` yet.

**Comment by akshatki** [comment]
@hawflau Thanks for taking a look and creating a pull request immediately, I see that the pull request is now merged. Although the following issues still remain:

1. When I launch the template [1] from the Cloudformation console, I still run into this error:
"Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [] is invalid. property OwnershipControls not defined for resource of type AWS::S3::Bucket. Rollback requested by user."

2. If I run the template [2] on the console, stack creates successfully. 

This hints that although OwnershipControls property is now supported within SAM but somewhere the workflow does not allow referencing this bucket in the events property of 'AWS::Serverless::Function' resource.

I have tested another testcase where I tried referencing this bucket being created in the template within the Outputs section of the template but that worked as stack entered CREATE_COMPLETE in this case. Template for this case [3].

This hints that as the 'events' property creates a AWS::Lambda::Permission resource and sets the NotificationConfiguration under AWS::S3::Bucket resource, the issue only happens when referencing the bucket inside the event property in AWS::Serverless::Function resource.

[1] https://gist.github.com/akshatki/61765ca285ce6c7893f2ebda070e6862

[2] https://gist.github.com/akshatki/0cee099126303c4b043d95d54a40c5bb

[3] https://gist.github.com/akshatki/207208625e74e5583cdce45cb0ff62ba
